entity/contents/filters/filter_input.cc

```diff
@@ -27,63 +51,118 @@ FilterInput::Vector FilterInput::Make(std::initializer_list<Variant> inputs) {
   return result;
 }
 
-FilterInput::Variant FilterInput::GetInput() const {
-  return input_;
+Matrix FilterInput::GetLocalTransform(const Entity& entity) const {
+  return Matrix();
 }
 
-std::optional<Rect> FilterInput::GetCoverage(const Entity& entity) const {
-  if (snapshot_) {
-    return snapshot_->GetCoverage();
-  }
+Matrix FilterInput::GetTransform(const Entity& entity) const {
+  return entity.GetTransformation() * GetLocalTransform(entity);
+}
 
-  if (auto contents = std::get_if<std::shared_ptr<Contents>>(&input_)) {
-    return contents->get()->GetCoverage(entity);
-  }
+FilterInput::~FilterInput() = default;
 
-  if (auto texture = std::get_if<std::shared_ptr<Texture>>(&input_)) {
-    return entity.GetPathCoverage();
-  }
+/*******************************************************************************
```

Comment:
In a later patch if necessary, you could toss these in their own TUs (perhaps in a "filters" directory).

Reply:
Ah yeah, I'll do this in a follow-up since https://github.com/flutter/impeller/pull/145 changes a bunch of stuff in `FilterInput`.